signer/src/crypto/secret_uri.rs

```diff
@@ -115,10 +117,10 @@ impl std::str::FromStr for SecretUri {
 }
 
 /// This is returned if `FromStr` cannot parse a string into a `SecretUri`.
-#[derive(Debug, Copy, Clone, PartialEq, thiserror::Error)]
+#[derive(Debug, Copy, Clone, PartialEq, Display)]
 pub enum SecretUriError {
     /// Parsing the secret URI from a string failed; wrong format.
-    #[error("Invalid secret phrase format")]
+    #[display(fmt = "Invalid secret phrase format")]
     InvalidFormat,
 }
 
```

Comment:
I left another comment about this but resolved it to start fresh:

The `once_static!` usage below returns `&'static regex::Regex` in "std" and `regex::Regex` in no-std. This is hidden in the `once_static!` macro. Being internal code, the type difference doesn't matter asl ong as it compiles in both cases anyway. What I do think is scary is that the `once_static!` macro hides this change, and in the future we might use it on something else (something public maybe) and forget that it emits different type signs in std and no-std.

I looked into `regex` a bit, and it already has had effort put into optimising its own Clone impl with internal ref counters etc.

So, let's just use `once_static_cloned` here, which has a nice identical-type-signature "no-std" fallback, and delete the `once_static` macro entirely!